runtime/android-driver/src/main/java/com/squareup/sqldelight/android/AndroidDatabaseHelper.kt

```diff
@@ -20,16 +19,42 @@ import com.squareup.sqldelight.db.SqlPreparedStatement.Type.SELECT
 import com.squareup.sqldelight.db.SqlPreparedStatement.Type.UPDATE
 import com.squareup.sqldelight.db.SqlResultSet
 
-class AndroidSqlDatabase(
-  private val openHelper: SupportSQLiteOpenHelper
+class AndroidSqlDatabase private constructor(
+  private val openHelper: SupportSQLiteOpenHelper? = null,
+  private val database: SupportSQLiteDatabase? = null
 ) : SqlDatabase {
+  constructor(
+    openHelper: SupportSQLiteOpenHelper
+  ) : this(openHelper = openHelper, database = null)
+
+  @JvmOverloads constructor(
+    helper: SqlDatabase.Helper,
+    context: Context,
+    name: String? = null,
+    callback: SupportSQLiteOpenHelper.Callback = AndroidSqlDatabase.Callback(helper)
+  ) : this(
+      database = null,
+      openHelper = FrameworkSQLiteOpenHelperFactory()
```

Comment:
I think it would make sense to make the factory a constructor arg that defaults to `FrameworkSQLiteOpenHelperFactory()`. That way you'd support alternative implementations of `androidx.sqlite`.

Reply:
Putting the factory into a default argument is fine, but you still have support for alternate implementations already with the other two constructors.

Reply:
I've somehow missed the `SupportSQLiteOpenHelper` constructor right above this one.

Reply:
I'd still prefer to make your change as well though!